0002/README.md

```diff
@@ -0,0 +1,167 @@
+---
+id: 0002
+title: Set the mood around releasing code
+status: ideation
+authors: Gianluca Arbezzano <gianarb92@gmail.com>
+---
+
+## Summary
+
+This proposal sets the expectation about what it means to release code part of
+the Tinkerbell organisation.
+
+## Goals and not Goals
+
+Goals are:
+
+1. What is a release
+2. Set a few rules that all the projects should follow
+3. Quality assertion.
+
+Not-Goals:
+
+1. How each project should be released
+2. Define who own releases
+3. Distribution via `apt`, `apk` and so on.
+
+## Content
+
+A release is a tested and stable point in time snapshot from a codebase.
+
+We follow the [Semantic Versioning (semver) 2.0.0](https://semver.org).
+Summarized as follows:
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+* MAJOR version when you make incompatible API changes,
+* MINOR version when you add functionality in a backwards compatible manner, and
+* PATCH version when you make backwards compatible bug fixes.
+
+Additional labels for pre-release and build metadata are available as extensions
+to the MAJOR.MINOR.PATCH format.
+
+### Schedule
+
+Every subproject should be able to follow its own cadence in terms of how often
+or regularly a new release should be cut, as long as it is documented and
+shared with the community.
+
+A new `MINOR` version for Tinkerbell should be released following a proper
+schedule, right now I think every two months sounds reasonable looking at the
+code written and merged in the last couple of weeks. In this way the
+subprojects will have an intrinsic release target if they want to get their new
+code included in the Tinkerbell release.
+
+`PATCH` release for Tinkerbell can be done as required by the subprojects
+bug fixes.
+
+`MAJOR` release right now are not even evaluated because the project is too young.
+
+### GitHub release and release website
+
+Git supports tag and GitHub has the concept of a release based on a particular
+tag.
+
+The artifact generated as part of the release will be published as part of the
+release website, using as a guidance how [HashiCorp
+Release](https://releases.hashicorp.com/) works.
+
+I like the idea to use a release website vs having to push them as manifest in
+GitHub release because it gives a single place where the community can
+programmatically lookup what they are looking for, without having to look for
+repositories, versions and so on.
+
+As part of the Release page on GitHub we will publish a generated changelog.
+
+### Architectures and operating systems
+
+We know there are different architectures and Go supports cross compilations.
+When possible we should cross compile binaries at least for:
+
+* Linux
+* macOS
+* Windows
+
+For `arm` and `x86_64`.
```

Comment:
Should this be `arm64` here?

Or are there also plans to support 32 bit arm?
- If so, which arm target for linux (`armhf` vs `armel`)? I'd assume `armhf`
- Which GOARM version? I'd assume `GOARM=7`